### ISSUE TITLE
Empty comment causes an exception in WrikeComment(string text) in WrikeComment.cs

### ISSUE THREAD

**Issue by snaddenm** [staviloglu/Taviloglu.Wrike.ApiClient#10]
This code throws an exception if text is an empty string:

```cs
public WrikeComment(string text)
{
    text.ValidateParameter(nameof(text));
    Text = text;
}
```

One way to generate an empty string is to add an attachment to Wrike without adding a comment, for example paste a bitmap into the comment box then submit it.

I have worked around this in my local working copy of the code as shown below. This change works for me, but I don't know if this has any other side-effects.

```cs
public WrikeComment(string text)
{
    // MJS Empty comments are actually OK, as they might just be a picture
    if (text != "")
        text.ValidateParameter(nameof(text));
    Text = text;
}
```

**Comment by snaddenm** [comment]
I should add that I'm using the latest code form the master branch and not the latest release, in order to be able to use an API v4 token.

**Comment by staviloglu** [comment]
Hi @snaddenm.

Thanks for sharing your problem.

WrikeComment constructor method throws exception if the text parameter is empty or null because of the [api documentation](https://developers.wrike.com/documentation/api/methods/create-comment) which says it can not be empty.

If you could share a code sample showing how you use attchments and comments together with api client that would be easier for me to understand your scenario. 

If you need to add attachment without a comment text, for example to a task, you can use [client.Attachments.CreateInTask](https://developers.wrike.com/documentation/api/methods/create-wrike-attachment) for that purpose. 

As far as I know, comments have attachmentId list but I can't find a way to add a comment with an attachment in the [api documentation](https://developers.wrike.com/documentation/api/methods/create-comment). I believe that is for the comments added using the user interface.

To sum up if you share your code we can discuss on it together and can find a better solution rather than a workaround.

**Comment by snaddenm** [comment]
I captured the call stack:

```
Taviloglu.Wrike.Core.dll!Taviloglu.Wrike.Core.Comments.WrikeComment.WrikeComment(string text) Line 19
Taviloglu.Wrike.Core.dll!Taviloglu.Wrike.Core.Comments.WrikeTaskComment.WrikeTaskComment(string text, string taskId) Line 8
Taviloglu.Wrike.ApiClient.WrikeClient.SendRequest<Taviloglu.Wrike.Core.Comments.WrikeTaskComment>(string requestUri, string httpMethod, System.Net.Http.HttpContent httpContent, Newtonsoft.Json.JsonConverter jsonConverter) Line 364
Taviloglu.Wrike.ApiClient.WrikeClient.Taviloglu.Wrike.ApiClient.IWrikeCommentsClient.GetInTaskAsync(Taviloglu.Wrike.ApiClient.WrikeClientIdParameter taskId, bool? plainText) Line 85
WrikeTracSync.exe!WrikeTracSync.Wrike.GetCommentsAync(Taviloglu.Wrike.Core.Tasks.WrikeTask task) Line 976
```

The last entry is a piece of my code which is:

```cs
// Get all the comments for a task and load them into the cache
public async Task<List<WrikeTaskComment>> GetCommentsAync(WrikeTask task) {
    Int32 attempts = 0;
    Boolean success = false;
    List<WrikeTaskComment> comments = null;
    while ((!success) && (attempts++ < 5)) {
        try {
            comments = await wrikeClient.Comments.GetInTaskAsync(taskId: task.Id);
            success = true;
        }
        catch (Newtonsoft.Json.JsonReaderException) {
            Console.WriteLine(" - Exception in Wrike.GetTasksAsync, retrying");
            Thread.Sleep(1000);
        }
    }
    if (!success)
        throw new Exception("Failed to read comments");
    if (!commentsByTask.ContainsKey(task.Id))
        commentsByTask.Add(task.Id, comments);
    else
        commentsByTask[task.Id] = comments;
    return comments;
}
```

Basically, the exception happens when I call Comments.GetInTaskAsync(taskId) to download the comments for a particular task. The task that causes the exception has a bitmap that was added by a user (Not me)  without typing any comment text.

I can reproduce this using the Wrike client app by:

1. Create a task
2. Copy a bitmap into the clipboard by taking a screenshot of something
3. Paste the bitmap into the comment field
4. Press the send button, without typing any text first
5. Now I have a comment which contains a bitmap and no text
6. Call Comments.GetInTaskAsync()

**Comment by staviloglu** [comment]
@snaddenm now I understand the problem clearly, thanks for your detailed comments. I wanted to enforce user to instantiate a comment object with a text parameter. But for your case I have added a private constructor for JSON.Net so it can still instantiate the WrikeComment object without text and fill its properties.
